src/test/javascript/spec/component/participation-submission/participation-submission.component.spec.ts

```diff
@@ -30,21 +30,37 @@ import { SubmissionExerciseType, SubmissionType } from 'app/entities/submission.
 import { StudentParticipation } from 'app/entities/participation/student-participation.model';
 import { TextSubmission } from 'app/entities/text-submission.model';
 import { RouterTestingModule } from '@angular/router/testing';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { HttpResponse } from '@angular/common/http';
+import { TemplateProgrammingExerciseParticipation } from 'app/entities/participation/template-programming-exercise-participation.model';
+import { ProgrammingSubmission } from 'app/entities/programming-submission.model';
+import { ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { SolutionProgrammingExerciseParticipation } from 'app/entities/participation/solution-programming-exercise-participation.model';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { ProfileInfo } from 'app/shared/layouts/profiles/profile-info.model';
+import { ParticipationService } from 'app/exercises/shared/participation/participation.service';
 
 chai.use(sinonChai);
 const expect = chai.expect;
 
 describe('ParticipationSubmissionComponent', () => {
     let comp: ParticipationSubmissionComponent;
     let fixture: ComponentFixture<ParticipationSubmissionComponent>;
+    let participationService: ParticipationService;
     let submissionService: SubmissionService;
+    let exerciseService: ExerciseService;
+    let programmingExerciseService: ProgrammingExerciseService;
+    let profileService: ProfileService;
     let findAllSubmissionsOfParticipationStub: SinonStub;
     let debugElement: DebugElement;
     let router: Router;
+    const route = { params: of({ participationId: 1, exerciseId: 42 }) };
 
     beforeEach(async () => {
         return TestBed.configureTestingModule({
-            imports: [ArtemisTestModule, NgxDatatableModule, ArtemisResultModule, ArtemisSharedModule, TranslateModule.forRoot(), RouterTestingModule],
+            imports: [ArtemisTestModule, NgxDatatableModule, ArtemisResultModule, ArtemisSharedModule, TranslateModule.forRoot(), RouterTestingModule, MomentModule],
```

Comment:
Can't any of these module mocked?

Reply:
what do you mean?

Reply:
I think he refers to mocking the individual components/directives instead of importing the module.